features/cellular/framework/AT/AT_CellularContext.cpp

```diff
@@ -659,17 +665,27 @@ nsapi_error_t AT_CellularContext::disconnect()
         // 3GPP TS 27.007:
         // For EPS, if an attempt is made to disconnect the last PDN connection, then the MT responds with ERROR
         if (_is_context_active && (rat < CellularNetwork::RAT_E_UTRAN || active_contexts_count > 1)) {
+            _at.clear_error();
```

Comment:
Return value is actually meaningless due to `clear_error` calls?

Reply:
almost, CGACT can still fail. I added those so that we try to disconnect as much as possible.